### DISCUSSION TITLE
Does one need to set a seed for plotBrowserTrack?

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#1642] [Questions / Documentation] [answered]
Hi @rcorces and @jeffmgranja ,

I noticed that the coverage scale for the bulk track plotting changes from run to run despite setting the ylim paramter to c(0, 0.999). 

For example, this is the result from run 1:

<img width="428" alt="Screen Shot 2022-09-23 at 4 01 06 PM" src="https://user-images.githubusercontent.com/54336249/192048529-82a5e132-3f56-4d58-9cd2-03ec0be09686.png">

And this is the result with the same parameters from run 2:

<img width="434" alt="Screen Shot 2022-09-23 at 4 07 36 PM" src="https://user-images.githubusercontent.com/54336249/192048847-ca8145cf-5d13-4925-967c-200b8fa05a98.png">

Aside from setting a seed, which seems to maintain reproducibility internally for myself, is there another strategy or fix I could implement here? 

Thanks for your help!

**Comment by rcorces** [comment, marked as answer]
This has been brought up previously and is due to randomized sub-sampling of cells when making plots. As the param docs say:

```
@param ylim The numeric quantile y-axis limit to be used for for "bulkTrack" plotting. This should be expressed as `c(lower limit, upper limit)` such as `c(0,0.99)`. If not provided, the y-axis limit will be c(0, 0.999).
```

Because the ylim is being set in quantile space, if the sub-sampling takes different cells, then the actual y limit of the plot will change.  Certainly on your end, you can fix this by setting a seed. I'm not familiar enough with randomized operations to know if this would be reproducible across computers as well. I guess if you really wanted it to be reproducible across computers, you could set `maxCells` to be the maximum number of cells in any group, effectively disabling sub-sampling. But this will have negative impacts on computation time. `maxCells` was introduced recently and I believe is only available on `dev` at the moment.

**Comment by RegnerM2015** [comment]
Thanks @rcorces! This answers my question.
